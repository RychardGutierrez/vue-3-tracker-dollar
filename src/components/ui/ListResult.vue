<template>
  <!-- search results  -->
  <ul v-if="countrySearchResults" class="absolute bg-dollar-secondary text-white w-full shadow-md py2 px1 top-[120px]">
    <p v-if="searchError.error" class="py-2 px-2">
      {{ searchError.message }}
    </p>

    <!-- message error-->
    <p v-if="!searchError.error && countrySearchResults.length === 0" class="py-2 px-2">
      Sorry, something went wrong, please try again
    </p>

    <template v-else>
      <li v-for="searchResult in countrySearchResults" :key="searchResult.name.common"
        @click="goToCountry(searchResult)"
        class="py-2 px-2 border-b border-y-slate-100 hover:bg-dollar-third cursor-pointer">
        {{ searchResult.name.official }}
      </li>
    </template>
  </ul>
</template>

<script setup>

defineProps({
  countrySearchResults: {
    typeof: Array,
    default: null
  },

  searchError: {
    typeof: Object,
    default: { error: false, message: '' }
  },

  goToCountry: {
    typeof: Function,
    default: () => { }
  }

})

</script>

<style lang="scss" scoped>

</style>