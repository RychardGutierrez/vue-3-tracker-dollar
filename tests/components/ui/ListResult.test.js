import { shallowMount } from '@vue/test-utils';
import ListResult from '../../../src/components/ui/ListResult.vue';

describe('ListResult component', () => {
  let wrapper = null;
  const jestGoToCountry = jest.fn();
  const props = {
    countrySearchResults: [
      {
        name: {
          common: 'Venezuela',
          official: 'Bolivarian Republic of Venezuela',
        },
      },
      {
        name: {
          common: 'Bolivia',
          official: 'Plurinational State of Bolivia',
        },
      },
    ],
    searchError: { error: false, message: '' },
    goToCountry: jestGoToCountry,
  };

  beforeEach(() => {
    wrapper = shallowMount(ListResult, { props });
  });

  test('should to match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should show an error message', () => {
    wrapper = shallowMount(ListResult, {
      props: {
        ...props,
        searchError: { error: true, message: 'Ups something is wrong' },
      },
    });

    const messageError = wrapper.get('p');
    expect(messageError.text()).toBe('Ups something is wrong');
  });

  test('should show an error message when we do not have result', () => {
    wrapper = shallowMount(ListResult, {
      props: {
        ...props,
        countrySearchResults: [],
      },
    });

    const messageError = wrapper.get('p');
    expect(messageError.text()).toBe(
      'Sorry, something went wrong, please try again'
    );
  });
});
