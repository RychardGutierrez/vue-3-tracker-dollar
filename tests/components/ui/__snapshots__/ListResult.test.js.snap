// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListResult component should to match with snapshot 1`] = `
"<!-- search results  -->
<ul class=\\"absolute bg-dollar-secondary text-white w-full shadow-md py2 px1 top-[120px]\\">
  <p class=\\"py-2 px-2\\"></p><!-- message error-->
  <li class=\\"py-2 px-2 border-b border-y-slate-100 hover:bg-dollar-third cursor-pointer\\">Bolivarian Republic of Venezuela</li>
  <li class=\\"py-2 px-2 border-b border-y-slate-100 hover:bg-dollar-third cursor-pointer\\">Plurinational State of Bolivia</li>
</ul>"
`;
